com.unity.render-pipelines.core/Runtime/Common/DynamicResolutionHandler.cs

```diff
@@ -147,58 +232,84 @@ public void SetCurrentCameraRequest(bool cameraRequest)
+                s_ActiveInstance = s_DefaultInstance;
+                newCameraId = 0;
+            }
+            else
+            {
+                s_ActiveInstance = GetOrCreateDrsInstanceHandler(camera);
+                newCameraId = camera.GetInstanceID();
+            }
+
+            s_ActiveInstanceDirty = newCameraId != s_ActiveCameraId;
+            s_ActiveCameraId = newCameraId;
+            s_ActiveInstance.Update(settings.HasValue ? settings.Value : s_ActiveInstance.m_CachedSettings, OnResolutionChange);
+        }
+
         /// <summary>
         /// Update the state of the dynamic resolution system.
         /// </summary>
         /// <param name="settings">The settings that are to be used by the dynamic resolution system.</param>
         /// <param name="OnResolutionChange">An action that will be called every time the dynamic resolution system triggers a change in resolution.</param>
         public void Update(GlobalDynamicResolutionSettings settings, Action OnResolutionChange = null)
         {
+            GarbageCollectCameras();
             ProcessSettings(settings);
 
-            if (!m_Enabled) return;
+            if (!m_Enabled && !s_ActiveInstanceDirty)
+            {
+                s_ActiveInstanceDirty = false;
+                return;
+            }
 
             if (!m_ForcingRes)
             {
+                bool canUseCurrentResFn = m_DynamicResMethod != null;
+                PerformDynamicRes dynResFunction     = canUseCurrentResFn ? m_DynamicResMethod : s_DefaultInstance.m_DynamicResMethod;
+                DynamicResScalePolicyType scalerType = canUseCurrentResFn ? m_ScalerType : s_DefaultInstance.m_ScalerType;
+
                 if (m_ScalerType == DynamicResScalePolicyType.ReturnsMinMaxLerpFactor)
                 {
-                    float currLerp = m_DynamicResMethod();
+                    float currLerp = dynResFunction();
                     float lerpFactor = Mathf.Clamp(currLerp, 0.0f, 1.0f);
                     m_CurrentFraction = Mathf.Lerp(m_MinScreenFraction, m_MaxScreenFraction, lerpFactor);
                 }
                 else if (m_ScalerType == DynamicResScalePolicyType.ReturnsPercentage)
                 {
-                    float percentageRequested = Mathf.Max(m_DynamicResMethod(), 5.0f);
+                    float percentageRequested = Mathf.Max(dynResFunction(), 5.0f);
                     m_CurrentFraction = Mathf.Clamp(percentageRequested / 100.0f, m_MinScreenFraction, m_MaxScreenFraction);
                 }
             }
 
-            if (m_CurrentFraction != m_PrevFraction)
-            {
-                m_PrevFraction = m_CurrentFraction;
-
-                if (!m_ForceSoftwareFallback && type == DynamicResolutionType.Hardware)
-                {
-                    ScalableBufferManager.ResizeBuffers(m_CurrentFraction, m_CurrentFraction);
-                }
-
-                if (OnResolutionChange != null)
-                    OnResolutionChange();
```

Comment:
This was called for SW dynamic res too and it doesn't look like it is called anymore?

Reply:
yep. nice catch. Restored.